results_paper/results.ipynb

```diff
@@ -0,0 +1,1237 @@
+    "#         violins(contestant, one_target, 'objids', label_dict, color='#00A878')\n",
+    "        violins(contestant, one_target, 'object_id', sub_labels, color=color_dict[contestant])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for contestant in ['validation']:#contestants:\n",
+    "    for one_target in label_dict.keys():#truth.true_target.unique():#label_dict:\n",
+    "#         violins(contestant, one_target, 'objids', label_dict, color='#00A878')\n",
+    "        violins(contestant, one_target, 'object_id', label_dict, color=color_dict[contestant])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "At least for Kyle's submission, classes 6, 15, 16, 53, 64, 65, 88, look pretty darn good, like what we'd expect from the \"perfect\" classifier archetype; class 90 looks more like the \"almost perfect\" or \"noisy\" classifier archetypes.\n",
+    "Classes 42 and 62 look like the \"mutually subsuming\" classifier archetype, relative to class 99; class 67 looks like a weaker form of the \"mutually subsuming\" classifier archetype relative to classes 62, 90, and 99, and class 95 is also like that, relative to only class 99.\n",
+    "Meanwhile, class 99 looks like the \"mutually subsuming\" classifier archetype relative to classes 42 and 62.\n",
+    "Class 52 looks like the \"uncertain\" classifier archetype, with respect to classes 42, 62, 67, 90, and 99.\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Next steps\n",
+    "\n",
+    "TODO: Consider making one plot per predicted class, which sort of conveys a probabilistic notion of false positives, whereas making one plot per true class sort of conveys a probabilistic notion of false negatives.\n",
+    "~~This would require splitting up the data files quite differently.~~\n",
+    "\n",
+    "Besides all the \"TODO\" items, I also want to try some unsupervised clustering on the probability vectors (per true class) to get an idea of the covariances, at least for the weirdos like 42, 52, 62, 67.\n",
+    "Actually, we know why this is happening!\n",
+    "It's due to the way Kyle generated the probability of being in class 99, a formula that actually uses the probabilities of these classes.\n",
+    "In a sense it's a bug that it draws probabilty away from true 42 (SNII) to a label of 99 (other)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# SCRATCH"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "still don't know why this cell is so slow"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# # Run me only if you've never run the notebook before, or if you changed the probabilities at which to evaluate the KDEs.\n",
```

Comment:
Should we include some sort of "first_run" flag to do these cells automatically but not have them commented?